src/compat/posix/include/pthread.h

```diff
@@ -60,6 +60,22 @@ typedef struct pthread_attr {
 	struct sched_param sched_param;
 } pthread_attr_t;
 
+struct _pthread_barrier_t {
+    unsigned int in;
+    unsigned int round;
+    unsigned int count;
+    int istep;
+    int pshared;
+    int initialize_flag;
+    int in_use;
+    sem_t lock[2];
```

Comment:
I don't quite understand why do we need two locks. Isn't 1 semaphore enough?

Reply:
Two semaphores are used to avoid race condition in between semaphore wait and post in case of multiple rounds.
Supposing, in a program, we initialize [count](https://github.com/sobhaniiest/embox/blob/7b445551739aedf49ee5812286ddc45553c92f7a/src/compat/posix/pthread/pthread_barrier.c#L38) by 3 and we are calling `pthread_barrier_wait` 6 times, so there will be 2 rounds. In the first round, when the 3rd `pthread_barrier_wait` will be called, it will go for [sem_post_multiple](https://github.com/sobhaniiest/embox/blob/7b445551739aedf49ee5812286ddc45553c92f7a/src/compat/posix/pthread/pthread_barrier.c#L128) and at that time if 4th `pthread_barrier_wait` is called then we don't want to use the same semaphore as it is in use (sem_post for the threads in 1st round). [Here](https://github.com/sobhaniiest/embox/blob/7b445551739aedf49ee5812286ddc45553c92f7a/src/compat/posix/pthread/pthread_barrier.c#L110) the value of `ib->istep` is alternated to use 2nd semaphore in the second round.

Reply:
I've also added a test for multiple rounds. [Here](https://github.com/sobhaniiest/embox/blob/7b445551739aedf49ee5812286ddc45553c92f7a/src/tests/posix/pthread/pthread_barrier_test.c#L65)